### ISSUE TITLE
Flies

### ISSUE THREAD

**Issue by ghost** [soulkobk/ArmA_Scripts#1]
Hey,

Nice feature! but one question, so each time there is a dead body the server creates a new thread (spawn) ?

 If so ouch that will hit performance

**Comment by soulkobk** [comment]
Yes it spawns a single thread per dead body (player or AI) for the duration the dead body is still a valid object (allDeadMen).

I'm not sure what sort of performance hit it will have on a highly populated server, as I have not tested within that scenario yet.

In my MP testing on a group of 10 or so AI, I didn't notice any performance hit... it will be interesting to see how the script performs when there are 50+ dead bodies at any one time.

Maybe you'd like to test it and report back?

EDIT*...

I tested in a MP scenario with 50-70 dead AI at any-one-time, and didn't notice any performance hit. I then tested the same scenario with ~120 dead AI, and did notice a performance hit of around ~10fps.

I'm not sure what A3WL server would have 100+ dead bodies on the map at any-one-time though.

If there is an issue with the script performance, one solution would be to lessen the time of the engine clean up via "corpseRemovalMinTime" and "corpseRemovalMaxTime" entries in the mission description.ext file.

**Comment by ghost** [comment]
I have 10min body clean up and it generally runs at 120-180 bodies on average depending how busy it is (stratis is mad) and a server fps on average of 15 after 4 hours.

Shame as I like the immersion this would give.

**Comment by soulkobk** [comment]
This was a script I started months ago... and finally finished it (after some testing). I like the immersion factor also, hence why I wrote it.

From brief tests, it's the particles (flies) that seem to slow the frames, but that's how ArmA has always been... just throw a heap of smoke grenades, or explode some vehicles... and the particles of the fire and smoke decrease frames straight away. Until BIS fix/enhance their particle system, it will always be like that unfortunately.

I also tested with a 10 minute dead body counter (corpseRemovalMaxTime = "10*60").

Whatever the case, the script is there if end-users want to use it. :)
